enhancements/project-request-limit.md

```diff
@@ -0,0 +1,293 @@
+resource. This will sever the "legacy" link between the plugin and the `kube-apiserver`.
+* We can provide the customer with seamless install and automatic upgrades by shipping the operator via OperatorHub 
+  independent of OpenShift release cycle.   
+
+
+### Goals
+1. Enable the `requestlimit.project.openshift.io/ProjectRequestLimit` admission plugin.
+2. Use existing extension points, libraries, and installation mechanisms in the manner we would recommend to external teams.
+3. Have a fairly straightforward way to install and enable this admission plugin.
+4. `rebootstrapping` must be possible.
+
+### Non-Goals
+1. Revisit how `requestlimit.project.openshift.io/ProjectRequestLimit` works. We want to lift it as is.
+2. Couple a slow moving admission plugin to a fast moving `kube-apiserver`.
+
+### Open Questions
+1. How do we productize the operator? Since this operator is not part of OpenShift core, it can essentially be built and
+ pushed to OperatorHub independently of OpenShift release cycle.
+3. How do we describe version skew limitations to OLM so our operator gets uninstalled *before* an illegal downgrade or
+ upgrade? This is a concrete case of the API we want to use isn't available before 1.16 and after 1.18, the previous API could be gone.
+
+## Proposal
+1. Create a `Validating` admission webhook server that provides `requestlimit.project.openshift.io/ProjectRequestLimit`.
+2. The admission webhook will enable `localhost` serving and `kube-apiserver` will talk to it using a `localhost` URL on the host network.
+2. Create an operator that manages all lifecycle aspects of the admission webhook via a custom resource.
+3. Productize the operator and ship it as a `RedHat operator` via `OperatorHub`. 
+4. Provide official documentation on how to interact with the `ProjectRequestLimit` operator.
+
+### User Stories [optional]
+
+*Story 1*: As an engineer I want to port `ProjectRequestLimit` plugin from 3.11 into a `Validating` admission webhook. 
+
+Notes:
+* Use  [generic-admission-server](https://github.com/openshift/generic-admission-server) to wire the validating admission webhook server.
+* The source code is here: https://github.com/openshift/origin/blob/release-3.11/pkg/project/apiserver/admission/requestlimit/admission.go 
+
+*Story 2*: As an engineer I want to write an operator that manages the `ProjectRequestLimit` admission webhook server.
+
+*Story 3*: As an engineer I want to ship `ProjectRequestLimit` as a RedHat operator via OperatorHub.
+
+Notes:
+* The operator is OLM enabled.
+* For productization, if we go through ART, integration work should target 4.5.
+* The operator should support disconnected install.
+  
+
+### Implementation Details/Notes/Constraints [optional]
+We *must* be able to re-bootstrap the cluster. This means that a cluster with this admission plugin created must be able
+to be completely shut down and subsequently restarted.
```

Comment:
so "rebootstrappable" is "rebootable" ?

Reply:
`rebootstrapping` no longer applies to this webhook, it's an openshift-apiserver webhook.